pkg/consumer/consumer.go

```diff
@@ -17,50 +19,36 @@ type Consumer interface {
 	Next() (library.Package, error)
 }
 
-type SQSConsumer struct {
-	client   *sqs.SQS
-	queueURL *string
+type PubSubConsumer struct {
+	subs *pubsub.Subscription
+	url  string
 }
 
-func NewSQSConsumer(sess *session.Session, queue string) (*SQSConsumer, error) {
-	client := sqs.New(sess)
-	result, err := client.GetQueueUrl(&sqs.GetQueueUrlInput{
-		QueueName: &queue,
-	})
+func NewPubSubConsumer(ctx context.Context) (*PubSubConsumer, error) {
+	queueURL := os.Getenv("OSSMALWARE_QUEUE_URL")
+	subs, err := pubsub.OpenSubscription(ctx, queueURL)
 	if err != nil {
 		return nil, err
 	}
-	consumer := &SQSConsumer{
-		client:   client,
-		queueURL: result.QueueUrl,
-	}
-	return consumer, nil
+	return &PubSubConsumer{
+		url:  queueURL,
+		subs: subs,
+	}, nil
 }
 
-func (c *SQSConsumer) Next() (library.Package, error) {
-	log.Printf("attempting to fetch message from %s", *c.queueURL)
-	msgResult, err := c.client.ReceiveMessage(&sqs.ReceiveMessageInput{
-		QueueUrl:            c.queueURL,
-		MaxNumberOfMessages: aws.Int64(1),
-		VisibilityTimeout:   aws.Int64(60 * 30),
-	})
-
+func (c *PubSubConsumer) Next() (library.Package, error) {
+	log.Printf("attempting to fetch message from %s", c.url)
+	msgResult, err := c.subs.Receive(context.Background())
 	if err != nil {
 		return library.Package{}, err
 	}
 
-	if len(msgResult.Messages) == 0 {
-		return library.Package{}, ErrEmpty
-	}
-
 	pkg := library.Package{}
-	msg := msgResult.Messages[0]
-	json.Unmarshal([]byte(*msg.Body), &pkg)
-
-	c.client.DeleteMessage(&sqs.DeleteMessageInput{
-		ReceiptHandle: msg.ReceiptHandle,
-		QueueUrl:      c.queueURL,
-	})
+	msg := msgResult.Body
+	if err := json.Unmarshal(msg, &pkg); err != nil {
+		return library.Package{}, err
+	}
+	msgResult.Ack()
```

Comment:
Do we need to bump this up to be after the call to `c.subs.Receive`? Essentially, making this a `defer msgResult.Ack()`, that way if json unmarshaling fails for some reason we'll still ack.

Reply:
I actually left it here on purpose, in case there was some intermittent error the lack of an ack would mean it would get picked up again.

But, since the only thing that can really fail is the json unmarshal we probably don't want to retry that forever...